corehq/apps/geospatial/views.py

```diff
@@ -435,3 +439,110 @@ def get_users_with_gps(request, domain):
+        if len(case_id_to_owner_id) > self.MAX_REASSIGNMENT_REQUEST_CASES:
+            return HttpResponseBadRequest(
+                _("Maximum number of cases that can be reassigned is {limit}").format(
+                    limit=self.MAX_REASSIGNMENT_REQUEST_CASES
+                )
+            )
+
+        if include_related_cases:
+            request_cases_id = list(case_id_to_owner_id.keys())
+            parent_to_child_cases_id = self.get_child_cases(domain, request_cases_id)
+            for parent_case_id, child_cases_ids in parent_to_child_cases_id.items():
+                for child_case_id in child_cases_ids:
+                    self._add_related_case(case_id_to_owner_id, parent_case_id, child_case_id)
+
+            child_to_parent_case_id = self.get_parent_cases(domain, request_cases_id)
+            for child_case_id, parent_case_id in child_to_parent_case_id.items():
+                self._add_related_case(case_id_to_owner_id, child_case_id, parent_case_id)
+
+        if len(case_id_to_owner_id) > self.ASYNC_CASES_LIMIT:
+            return HttpResponseBadRequest(
+                _("Maximum limit for cases to be reassigned including related cases exceeded."
+                  " Please select a lower value to update at time or consider deselecting 'include related cases'."
+                  " Reach out to support if the issue still persists.")
+            )
+
+        if len(case_id_to_owner_id) <= self.SYNC_CASES_UPDATE_THRESHOLD:
+            update_cases_owner(domain, case_id_to_owner_id)
+            return JsonResponse({'success': True, 'message': _('Cases were reassigned successfully')})
+        else:
+            return self.process_as_async(case_id_to_owner_id)
+
+    def process_as_async(self, case_id_to_owner_id):
+        task_id = f'geo_cases_reassignment_update_owners_{self.domain}'
+        if is_task_invoked_and_not_completed(task_id):
+            return HttpResponseBadRequest(
+                _('A case reassignment task is currently in progress. Please try again after some time')
+            )
+
+        geo_cases_reassignment_update_owners.apply_async((self.domain, case_id_to_owner_id), task_id=task_id)
+        return JsonResponse(
+            {
+                'success': True,
+                'message': _('Case reassignment request has been accepted and will be completed in some time')
+            }
+        )
+
+    def _add_related_case(self, case_id_to_owner_id, case_id, related_case_id):
+        if related_case_id not in case_id_to_owner_id:
+            case_id_to_owner_id[related_case_id] = case_id_to_owner_id[case_id]
+
+    def _format_as_list(self, data):
+        if isinstance(data, dict):
+            data = [data]
+        return data
+
+    def _get_parent_index(self, doc):
+        return next((index for index in doc[INDICES_PATH] if index[IDENTIFIER] == 'parent'), None)
+
+    def get_child_cases(self, domain, case_ids):
```

Comment:
it looks lie a lot is happening in this function and `get_parent_cases`.
Can you add some documentation to these please?

Reply:
Addressed in 9462200ac77ec093c46c66272baf3b10b96ce6b3